### PR TITLE
Use es6 modules and webpack in prep for NPM publish

### PR DESCRIPTION
**Pull request by jaimedp**
A lot of changes. I tried to do the messy change of removing and re-indenting files on the first commit and then fixes on the following commits to make it a bit easier to see, but it is still hard to see the changes :(

### CODE REVIEW

**Comment by narenranjit** [comment]
Great! Adding `?w=1` to github diff url ignores whitespace so wasn't too bad to see changes :) Few thoughts:

- Maybe I missed it but are we not exposing `nwt` publicly? Some examples like (http://forio.com/contour/gallery.html#/chart/scatter/scatter-trendline) depend on a few functions from that so we should?
- We should probably export it as `Contour.utils` instead?
- Can we replace the `lodash` functions in contour-utils with the actual lodash imports? We can import those specific functions without bundling all of lodash.

**Comment by narenranjit** [comment]
Also, in the spirit on modernizing jshintrc -> eslintrc ?

File: src/scripts/core/axis/axis-scale-factory.js [hunk 1]
```diff
@@ -1,63 +1,69 @@
...
+import CenteredYAxis from './centered-y-axis';
+import LinearScale from './linear-scale-axis';
+import LogYAxis from './log-y-axis';
+import OrdinalScale from './ordinal-scale-axis';
+import SmartYAxis from './smart-y-axis';
+import TimeScale from './time-scale-axis';
 
+
+export const axes = {};
+export const addAxis = (name, axisCtor) => {
```
Comment (narenranjit) [inline src/scripts/core/axis/axis-scale-factory.js]: Any reason not to `export function addAxis` instead? afaik export a const arrow function will show up as a anonymous in stacktraces, while exporting named functions will list it out.

**Comment by narenranjit** [comment]
What're the before/after file-sizes of contour.min.js?

File: src/scripts/core/axis/axis-scale-factory.js (hunk 1 above)
Comment (jaimedp) [reply src/scripts/core/axis/axis-scale-factory.js]: I don't think it makes any difference for the stacktrace, I test all the options and the traces still get the correct function name... then found this
http://lukecod.es/2015/12/01/es6-arrow-function-stack-traces/

**Comment by jaimedp** [comment]
- I added `Contour.utils` to expose nwt to the rest of the world.
- I think we can keep the custom lodash replacements for now..  I think breaking up with lodash should be complete :)
- I have another branch with eslint and all the fixes for linting but I think we should finish and merge this branch first
- The size of the new min.js file is ~82k, the previous one was ~78k, so is ~4K larger for the stuff that webpack injects I guess

Also, for publishing to npm, you'll need to create an npm user for forio and make sure the `author` info in `package.json` matches with the user, so we can publish it.

**Comment by narenranjit** [comment]
Looks good! 

- Lodash is overkill for the `isObject` etc of course, but why not use it for the `merge`, `defaults` etc? The less code under our maintenance purview and the more we handoff to established libraries the better, unless there's a reason not to?

- We should whitelist IE11 in preset-env.

**Comment by jaimedp** [comment]
- Lodash... Yeah, it makes sense.
- IE11... I was thinking the same thing last night, I guess I forgot to do it :)

One more thing to note. I removed the /dist folder from the repo, this means that you will no longer be able to use `bower install` since it does not have a post-install build. Just to make sure you're ok completely dropping support for `bower`.

**Comment by narenranjit** [comment]
I'd forgotten bower existed till I tried to build contour last week :)

Tangentially, we've given up on making millions with Contour-advanced so contents of that repo should go somewhere public, just not sure where. Thoughts? Just new showcase examples? (I'll get you access to the advanced repo in a bit).

**Comment by jaimedp** [comment]
Moved to use lodash.merge and lodash.defaults (increase in size of the minified file by ~13k). Also added ie11 to the env in babel and fixed 267 & 268.

Regarding advance viz, I think it would be good if they had their place in te showcase and then you could just install them like `npm install contour.drawable` or something like that

**Comment by narenranjit** [comment]
I'm having second thoughts about not including `dist`, though more about workflow than anything technical. 

For new PRs it's useful to have devs include a jsfiddle in their example pointing to a jsfiddle demoing their change, for e.g. see https://github.com/forio/contour/pull/251 which has http://jsfiddle.net/3yqyvdud/4/ which points to a build contour file for that PR. This really reduces the friction of testing PRs, and puts the onus of making sure it builds/works on the submitter.

Thoughts/ suggestions for a better workflow?

**Comment by jaimedp** [comment]
Hey, I see the point of evaluating PRs from a jsfiddle, but I think you'd end up with unnecessary  merge conflicts  which is minor, but the bigger problem I think it is that you end up with a very confusing versioning, since the /dist folder no longer contains the "official" released version, it has what ever the latest merge put there if it didn't forget to build it... for example in cases where the PR does not need to be js tested (ie. changes to less files or docs), then the /dist folder is not even the latest/edge version, is just a random snapshot.

There is one posible solution, but it would mean to drop the bundle of the lodash merge/default functions. The solution is to include the `/src/scripts/index.js` file as a `<script type="module">` that way (with some tweaks) it works, but it limits to `import` only files addressable from the src folder (ie. no node_modules).

(I did a proof of concept here)[http://jsfiddle.net/f48123x1/1/]

**Comment by narenranjit** [comment]
You can prevent merge conflicts / large diffs on dist if you mark minified files as "binary" in gitattributes (see https://github.com/forio/flow.js/blob/master/.gitattributes for e.g. https://blog.andrewray.me/dealing-with-compiled-files-in-git/ for more info).

Source type module is an interesting idea, let me play around with that.

**Pull request merged**